core/src/main/java/com/graphhopper/util/EdgeIterator.java

```diff
@@ -45,6 +45,11 @@
      */
     int NO_EDGE = -1;
 
+    /**
+     * integer value used to indicate that any edge is accepted
+     */
+    int ANY_EDGE = Integer.MIN_VALUE;
```

Comment:
what is a bit ugly here is that `EdgeIterator.Edge.isValid` only checks if an edge id is positive or not. this does not allow us to have other special values like this one here. maybe we should make `isValid` check if `==-1` ?

Reply:
Yeah, this is a good idea. Additionally we should have a look if we use `==-1` instead of `isValid` call. Have created #1666